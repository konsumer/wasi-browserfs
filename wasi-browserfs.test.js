import { setup, getFs, Buffer } from '.'
import { readFile } from 'fs/promises'

const fs = await getFs({
  fs: 'ZipFS',
  options: {
    zipData: Buffer.from(await readFile('example/simple.zip'))
  }
})

test('simple', async () => {
  const stdout = []
  const stderr = []
  const handleStdErr = t => stderr.push(t)
  const handleStdOut = t => stdout.push(t)

  const wasi_snapshot_preview1 = setup(fs, handleStdErr, handleStdOut)
  const { instance } = await WebAssembly.instantiate(await readFile('example/simple.wasm'), { wasi_snapshot_preview1 })
  wasi_snapshot_preview1._set_instance(instance)
  wasi_snapshot_preview1._stdin = 'here is some test text\n'
  instance.exports._start()
  expect(stdout).toMatchSnapshot()
  expect(stderr).toMatchSnapshot()
})
