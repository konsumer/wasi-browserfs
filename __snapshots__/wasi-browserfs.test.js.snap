// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`simple 1`] = `
[
  "This is a test of stderr",
  ":",
  " ",
  "Success",
  "
",
  "File Read Error",
  ":",
  " ",
  "Bad file descriptor",
  "
",
]
`;

exports[`simple 2`] = `
[
  "This is a simple tester of filesystem stuff.
",
  "
",
  "Filesize: 2818

",
  "finished reading file.
",
  "Please enter a line:
",
  "You entered , which has 22 chars.
",
]
`;
